packages/babel-plugin-import-jsx-pragma/test/index.js

```diff
@@ -50,6 +57,26 @@ describe( 'babel-plugin-import-jsx-pragma', () => {
 			isDefault: false,
 		} );
 
-		expect( string ).toBe( 'import { createElement } from "@wordpress/element";\nlet foo = <bar />;' );
+		expect( string ).toBe( 'import { createElement } from "@wordpress/element";\n' + original );
+	} );
+
+	it( 'adds import for scope variable even when defined inside the local scope', () => {
+		const original = 'let foo = <bar />;\n\nfunction local() {\n  const createElement = wp.element.createElement;\n}';
+		const string = getTransformedCode( original, {
+			scopeVariable: 'createElement',
+			source: '@wordpress/element',
+			isDefault: false,
+		} );
+
+		expect( string ).toBe( 'import { createElement } from "@wordpress/element";\n' + original );
```

Comment:
Actually nvm.  I understand this is technically okay assuming this kind of script was implemented in an environment which maps the import to the global.  It just sounds like its something we shouldn't automatically fix because this kind of thing would _likely_ be found in an environment where there _is_ no aliasing (and thus create a different kind of error in the built file for the developer).

Reply:
Well, I figured out that I took a very naive approach which won't work in a few cases. Your comment got me thinking what about IIFE which is a very common pattern in the WordPress world. For example:
https://github.com/WordPress/gutenberg/blob/master/packages/e2e-tests/plugins/plugins-api/sidebar.js#L12

I think that the implementation I proposed checks only variable bindings in the very outer scope.